packages/runtime/src/next/client.ts

```diff
@@ -563,13 +555,50 @@ export class Makeswift {
     return localizedGlobalElement
   }
 
-  async getPagePathnameSlice(pageId: string): Promise<PagePathnameSlice | null> {
-    const result = await this.graphqlClient.request<
-      PagePathnamesByIdQueryResult,
-      PagePathnamesByIdQueryVariables
-    >(PagePathnamesByIdQuery, { pageIds: [pageId] })
+  async getPagePathnameSlices(
+    pageIds: string[],
+    { preview: previewOverride = false, locale }: { preview?: boolean; locale?: string } = {},
+  ): Promise<(PagePathnameSlice | null)[]> {
+    const url = new URL(
+      `${this.siteVersion == null ? 'v1' : 'v2'}/page-pathname-slices/bulk`,
+      this.apiOrigin,
+    )
+
+    pageIds.forEach(id => url.searchParams.append('ids', id))
+    if (locale != null) url.searchParams.set('locale', locale)
+
+    const response = await this.fetch(url.pathname + url.search, {
+      headers: {
+        'Makeswift-Site-Version':
+          this.siteVersion ??
+          (previewOverride ? MakeswiftSiteVersion.Working : MakeswiftSiteVersion.Live),
+      },
+    })
+
+    if (!response.ok) {
+      console.error('Failed to get page pathname slices', await response.json())
+
+      return []
+    }
+
+    const json = await response.json()
+
+    const pagePathnameSlices = pagePathnameSlicesAPISchema.parse(json)
+
+    return pagePathnameSlices.map(({ pageId, pathname }) => ({
+      id: pageId,
+      pathname,
+      __typename: 'PagePathnameSlice',
+    }))
+  }
+
+  async getPagePathnameSlice(
+    pageId: string,
+    { preview = false, locale }: { preview?: boolean; locale?: string } = {},
+  ): Promise<PagePathnameSlice | null> {
+    const res = await this.getPagePathnameSlices([pageId], { preview, locale })
```

Comment:
nit: this is not a response object—it's page pathname slices

Reply:
Updated